### DISCUSSION TITLE
Tensor algebra

### DISCUSSION THREAD

**Discussion by hureje** [thelfer/tfel#213] [Q&A]
Let A and B be two second order tensors. How can I compute the product of B with the eigenvector associated with one eigenvalue (for example the largest) of tensor A ?

**Comment by thelfer** [comment]
Hi @hureje,

To be honest, I never had to compute the product of a second order tensors with a vector. But if you only need the result of v. (B *v), you can do:

```C++
  const auto [v, m] = A.computeEigenVectors();
  const auto [A1, A2, A3] = A.computeEigenTensors(m);
  const auto value = (B | A1);
```

Do you have more context on what you want to do ?
